CHANGELOG.md

```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Added
+
+- Adds a new template to generate metric semantic conventions, as well as adding it to the Makefile `semconv-generate` script to add the metric.go file to the v1.24.0 directory. (#4528)
```

Comment:
This will need to be changed to reflect the actual/final solution.

Reply:
Might not need to include the version information because I don't believe the metric.go file will be included in the final version of this PR